lib/runtime/life/resolver.go

```diff
@@ -424,26 +423,22 @@ func storageAppend(storage runtime.Storage, key, valueToAppend []byte) error {
 	if len(valueCurr) == 0 {
 		valueRes = valueToAppend
 	} else {
-		// remove length prefix from existing value
-		r := &bytes.Buffer{}
-		_, _ = r.Write(valueCurr)
-		dec := &scale.Decoder{Reader: r}
-		currLength, err := dec.DecodeBigInt() //nolint
-
+		var currLength *big.Int
+		err := scale.Unmarshal(valueCurr, &currLength)
 		if err != nil {
 			logger.Trace("[ext_storage_append_version_1] item in storage is not SCALE encoded, overwriting", "key", key)
 			storage.Set(key, append([]byte{4}, valueToAppend...))
 			return nil
 		}
 
-		// append new item
-		valueRes = append(r.Bytes(), valueToAppend...)
+		// append new item, pop off first byte now, since we're not using old scale.Decoder
+		valueRes = append(valueCurr[1:], valueToAppend...)
```

Comment:
Is there a risk that since currLength is a *big.Int that it will consume more space than the leading byte?

Reply:
why is the first byte popped off? is it because of the length encoding? the length encoding isn't necessarily 1 byte

Reply:
nice catch